evm/vm/message.py

```diff
@@ -46,6 +52,8 @@ def __init__(self,
                  data,
                  code,
                  origin=None,
+                 read_list=None,
+                 write_list=None,
```

Comment:
Is it valid to have only one of these lists present (read list but no write list or write list but no read list).  

If not, what do you think about changing this to be a single argument `read_and_write_list` which takes the form of a 2-tuple `(read_list, write_list)`.  Then within the constructor we can unpack it if it's not `None`:  `read_list, write_list = read_and_write_list`.